google/cloud/bigtable/internal/async_bulk_apply.cc

```diff
@@ -78,17 +71,33 @@ void AsyncRetryBulkApply::StartIterationIfNeeded(CompletionQueue cq) {
         self->OnRead(std::move(r));
         return make_ready_future(true);
       },
-      [self, cq](Status s) { self->OnFinish(cq, std::move(s)); });
+      [self, cq](Status const& s) { self->OnFinish(cq, s); });
 }
 
 void AsyncRetryBulkApply::OnRead(
     google::bigtable::v2::MutateRowsResponse response) {
   state_.OnRead(response);
 }
 
-void AsyncRetryBulkApply::OnFinish(CompletionQueue cq, Status status) {
-  state_.OnFinish(std::move(status));
-  StartIterationIfNeeded(std::move(cq));
+void AsyncRetryBulkApply::OnFinish(CompletionQueue cq, Status const& status) {
+  state_.OnFinish(status);
```

Comment:
I might have written this as:

```cc
void AsyncRetryBulkApply::OnFinish(CompletionQueue cq, Status status) {
  auto const is_retryable = status.ok() || rpc_retry_policy_->OnFailure(status);
  state_.OnFinish(std::move(status));
  if (!state_.HasPendingMutations() || !is_retryable) {
    promise_.set_value(std::move(state_).OnRetryDone());
    return;
  }
... ... ...  
```

Mostly because the copy of `Status` (which has a `std::string`) is more expensive than checking whether it is retryable.

Reply:
Done. but with `OnFinish(status)` because we need the value later in the method